src/Bicep.Core/Workspaces/SourceFileGrouping.cs

```diff
@@ -21,23 +42,29 @@ public record UriResolutionResult(
         bool RequiresRestore,
         ErrorBuilderDelegate? ErrorBuilder);
 
-    public record ModuleSourceResolutionInfo(
-        IForeignTemplateReference ForeignTemplateReference,
-        ISourceFile ParentTemplateFile);
-
     public record SourceFileGrouping(
         IFileResolver FileResolver,
         Uri EntryFileUri,
         ImmutableDictionary<Uri, FileResolutionResult> FileResultByUri,
         ImmutableDictionary<ISourceFile, ImmutableDictionary<IForeignTemplateReference, UriResolutionResult>> UriResultByModule,
         ImmutableDictionary<ISourceFile, ImmutableHashSet<ISourceFile>> SourceFileParentLookup) : ISourceFileLookup
     {
-        public IEnumerable<ModuleSourceResolutionInfo> GetModulesToRestore()
+        public IEnumerable<IArtifactResolutionInfo> GetModulesToRestore()
```

Comment:
Should the return type be left as is? Only `GetArtifactsToRestore` is mingling different kinds of `IArtifactResolutionInfo`.

Reply:
sure, that is not a bad idea